src/UnionFind/Monadic.hs

```diff
@@ -105,8 +105,9 @@ find ::
   (Ord key, Monad m) =>
   key ->
   WithUnionFindT key value m (key, Int, Maybe value)
-find key =
-  Map.lookup key <$> getBindings >>= \case
+find key = do
+  bindings <- getBindings
+  case Map.lookup key bindings of
     Nothing -> return (key, 1, Nothing)
```

Comment:
I don't really think this change improves readability. It avoids the two monadic operators on the same line but I think it was already reasonably understandable before, and actually a bit more concise (not that concision should be the goal).

Reply:
FYI, the change implemented here is not what `hlint` suggested, but it does make `hlint` happy. `hlint`'s suggestion was:
![image](https://user-images.githubusercontent.com/5920602/214537708-9c67f54b-0f5b-4766-ba9e-485c3a531f94.png)
which I find extremely ugly.

Reply:
FYI, the change implemented here is not what `hlint` suggested, but it does make `hlint` happy. `hlint`'s suggestion was:
![image](https://user-images.githubusercontent.com/5920602/214537708-9c67f54b-0f5b-4766-ba9e-485c3a531f94.png)
which I find extremely ugly.